crates/vm2/src/vm.rs

```diff
@@ -13,26 +13,24 @@ use crate::{
     ExecutionEnd, Instruction, ModeRequirements, Predicate, Program,
 };
 
-#[derive(Debug)]
+/// [`VirtualMachine`] settings.
+#[derive(Debug, Clone)]
 pub struct Settings {
     pub default_aa_code_hash: [u8; 32],
     pub evm_interpreter_code_hash: [u8; 32],
-
     /// Writing to this address in the bootloader's heap suspends execution
     pub hook_address: u32,
 }
 
+/// High-performance out-of-circuit ZKsync Era VM.
+#[derive(Debug)]
 pub struct VirtualMachine<T, W> {
-    pub world_diff: WorldDiff,
-
+    pub(crate) world_diff: WorldDiff,
     /// Storing the state in a separate struct is not just cosmetic.
     /// The state couldn't be passed to the world if it was inlined.
```

Comment:
I'd remove this comment. IIRC it is from a time when world was contained in the VM.